client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx

```diff
@@ -2,14 +2,19 @@ import { Button, Gridicon } from '@automattic/components';
 import classnames from 'classnames';
 import { translate, useRtl } from 'i18n-calypso';
 import Badge from 'calypso/components/badge';
-import { isTaskDisabled } from './task-helper';
+import { isTaskDisabled, hasIncompleteDependencies } from './task-helper';
 import { Task } from './types';
 
 const ChecklistItem = ( { task }: { task: Task } ) => {
 	const isRtl = useRtl();
 	const { id, isCompleted, keepActive, actionUrl, title, actionDispatch } = task;
 	const action = actionDispatch ? { onClick: actionDispatch } : { href: actionUrl };
 	const taskDisabled = isTaskDisabled( task );
+	const hasBadgeText = task.displayBadge && task.badgeText;
+
+	// Display chevron if task is incomplete. Don't display chevron and badge at the same time.
+	const shouldDisplayChevron =
+		! hasIncompleteDependencies( task ) && ! isCompleted && ! hasBadgeText;
```

Comment:
A bit of a nitpick - but Is there a reason to import and use `hasIncompleteDependencies` here, as opposed to leveraging the `taskDisabled` variable we already have above? Would `! taskDisabled && ! isCompleted && ! hasBadgeText` work here and be more readable?

Reply:
Sounds about right, it shouldn't be possible to have an incomplete task that's disabled

Reply:
> Is there a reason to import and use hasIncompleteDependencies here, as opposed to leveraging the taskDisabled variable we already have above?

`taskDisabled` returns false if `keepActive` is true, regardless of whether or not the task is complete.

https://github.com/Automattic/wp-calypso/blob/4344d9b1af04b09cf25f954ba193bff8beb97a08/client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts#L133-L145

This leads to the bug we're seeing above. Take the "Add links" task for example. `keepActive` is always true for the task, so `! taskDisabled` will also always return true, and a chevron will always be displayed, regardless of task completion status.

Reply:
Ah yes, I think the way you have it is better.  Thinking about it more what Im suggesting might have unexpected results in the future: a task that is meant to keep active, that we haven't met dependencies to enable yet, that is obviously not complete, etc... would end up showing a chevron. 😵 

While what you have in place for simply checking dependencies, completion state, and badge state seems like the most logically sound option here.

Reply:
Another note (not a blocker for this PR)- looking in that code snippet above for the `isTaskDisabled` function. Should we move the dependency check to be the keepActive check?  Right now it reads as if we have a task that is both meant to remain active after completion (keepEnabled), but also requires dependencies to be met before we first enable it, that it will always be enabled because of `keepEnabled` even if dependencies are not met?

Reply:
> Another note (not a blocker for this PR)- looking in that code snippet above for the isTaskDisabled function. Should we move the dependency check to happen before the keepActive check? Right now it reads as if we have a task that is both meant to remain active after completion (keepEnabled), but also requires dependencies to be met before we first enable it, that it will never be disabled because of keepEnabled even if dependencies are not yet met?

Makes sense to me 👍 . I'll draft a follow-up right after I merge this PR.

Reply:
https://github.com/Automattic/wp-calypso/pull/68171